method.rb

```diff
@@ -0,0 +1,71 @@
+module Enumerable
+    def my_each
+      for i in 0...self.length
+        yield(self[i])
+      end
+    end
+  
+    def my_each_with_index
+      k = 0
+      for i in 0...self.length
+        yield(self[i], k)
+        k += 1
+      end
+    end
+  
+    def my_select(&block)
+      result = []
+      self.each do |element|
+        result << element if block.call(element) == true
+      end
+      result
+    end
+  
+    def my_all?
```

Comment:
#### `REQUIRED`

:bug: There's a bug to fix. Let's add cases for when a `block` is not given and a `pattern` is passed as an `arg`. Compare with the built-in method.

- When a `block` is not given
```ruby
irb(main):003:0> [1,2,3,4].all?
=> true
irb(main):004:0> [1,2,3,4].my_all?
LocalJumpError (no block given (yield))
irb(main):005:0> 
```
- When a `pattern` (arg) is passed

```ruby
irb(main):005:0> [1,2,3,4].all?(2)
=> false
irb(main):006:0> [1,2,3,4].my_all?(2)
ArgumentError (wrong number of arguments (given 1, expected 0))
```

Learn more [here](https://ruby-doc.org/core-2.6.3/Enumerable.html)